pkg/cli/idle/idle.go

```diff
@@ -230,16 +231,16 @@ func scanLinesFromFile(filename string) ([]string, error) {
 	return lines, nil
 }
 
-// idleUpdateInfo contains the required info to annotate an endpoints object
+// idleUpdateInfo contains the required info to annotate a service object
 // with the scalable resources that it should unidle
 type idleUpdateInfo struct {
 	service   *corev1.Service
 	scaleRefs map[unidlingapi.CrossGroupObjectReference]struct{}
 }
 
 // calculateIdlableAnnotationsByService calculates the list of objects involved in the idling process from a list of services in a file.
-// Using the list of services, it figures out the associated scalable objects, and returns a map from the endpoints object for the services to
-// the list of scalable resources associated with that endpoints object, as well as a map from CrossGroupObjectReferences to scale to 0 to the
+// Using the list of services, it figures out the associated scalable objects, and returns a map from the endpointslices object for the services to
+// the list of scalable resources associated with that endpointslices object, as well as a map from CrossGroupObjectReferences to scale to 0 to the
```

Comment:
With Endpoints, the object name is plural, and there's guaranteed to be only one of them (hence, "the endpoints object"). With EndpointSlice, the object name is singular, but there may be more than one of them if the service has lots of endpoints or is dual-stack, so it should be "the endpointslice objects".

However, this also means the new code is slightly wrong; if there are multiple EndpointSlices for a Service, then you need to merge together the results of `findScalableResourcesForEndpoints` for each one together in the returned `endpointsInfo`, whereas the current version of this PR will overwrite the info calculated for the first EndpointSlice when it processes the second EndpointSlice.